packages/phone-number-privacy/common/src/utils/authentication.ts

```diff
@@ -66,16 +68,26 @@ export function verifyDEKSignature(
   registeredEncryptionKey: string,
   logger?: Logger
 ) {
+  logger = logger ?? rootLogger()
   try {
+    const msgDigest = crypto.createHash('sha256').update(JSON.stringify(message)).digest('base64')
+
     const key = ec.keyFromPublic(trimLeading0x(registeredEncryptionKey), 'hex')
     const parsedSig = JSON.parse(messageSignature)
-    return key.verify(message, parsedSig)
-  } catch (err) {
-    if (logger) {
-      logger.error('Failed to verify signature with DEK')
-      logger.error({ err, dek: registeredEncryptionKey })
+    if (key.verify(msgDigest, parsedSig)) {
+      return true
+    }
+
+    if (key.verify(message, parsedSig)) {
+      // TODO: Remove this once clients upgrade to @celo/identity v1.5.3
```

Comment:
We should be more explicit about why this code path exists and why it needs to be removed. Also, we should not write a `verify` function that exhibits a strange behavior like this without sufficient warning to the user. I suggest we add an (intentionally ugly) `insecureAllowIncorrectlyGeneratedSignature` parameter that should be set to `true` in order to activate this code path.
```suggestion
    // TODO: Remove this once clients upgrade to @celo/identity v1.5.3
    // Due to an error in the original implementation of the sign and verify functions
    // used here, older clients may generate signatures over the truncated message,
    // instead of its hash. These signatures represent a risk to the signer as they do
    // not protect against modifications of the message past the first 64 characters of the message.
    if (insecureAllowIncorrectlyGeneratedSignature && key.verify(message, parsedSig)) {
```